Main/src/main/java/com/main/acad/controller/Controller.java

```diff
@@ -1,37 +1,65 @@
 package com.main.acad.controller;
 
 import com.main.acad.annotation.MappingMethod;
-import com.main.acad.dao.ChapterDao;
-import com.main.acad.dao.Dao;
 import com.main.acad.entity.Chapter;
-import com.main.acad.serializator.MyJsonSerializer;
+import com.main.acad.serializator.JsonSerializatorImplementation;
+import com.main.acad.service.ChaptersServiceImplementation;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
-import java.io.PrintWriter;
 import java.util.List;
+import java.util.logging.Logger;
 
 public class Controller {
+
+    private static final Logger logger = Logger.getLogger(Controller.class.getName());
+
+    private static Controller instance;
+
+    public static Controller getInstance() {
+        if (instance == null) {
+            instance = new Controller();
```

Comment:
You can put field initialization logic (ChaptersServiceImplementation strategySimulationServiceImplementation = ChaptersServiceImplementation.getInstance();) there but it is not neccesery

Reply:
I do not use because in the servlet there is a logic to find a specific controller and in it there is a specific annotation of the method.He does not know what the controller will be

Reply:
And how that facts related?